cmd/xl-v1-metadata.go

```diff
@@ -47,91 +119,140 @@ func (t byObjectPartNumber) Len() int           { return len(t) }
-type HashAlgo string
+// MarshalJSON marshals the ChecksumInfo struct
+func (c ChecksumInfo) MarshalJSON() ([]byte, error) {
+	type checksuminfo struct {
+		Name string `json:"name"`
+		Hash string `json:"hash"`
+	}
 
-const (
-	// HashBlake2b represents the Blake 2b hashing algorithm
-	HashBlake2b HashAlgo = "blake2b"
-	// HashSha256 represents the SHA256 hashing algorithm
-	HashSha256 HashAlgo = "sha256"
-)
+	info := checksuminfo{
+		Name: c.Name,
+		Hash: hex.EncodeToString(c.Hash),
+	}
+	return json.Marshal(info)
+}
+
+// UnmarshalJSON unmarshals the the given data into the ChecksumInfo struct
+func (c *ChecksumInfo) UnmarshalJSON(data []byte) error {
+	type checksuminfo struct {
+		Name string `json:"name"`
+		Hash string `json:"hash"`
+	}
+
+	var info checksuminfo
+	err := json.Unmarshal(data, &info)
+	if err != nil {
+		return err
+	}
+	c.Hash, err = hex.DecodeString(info.Hash)
+	if err != nil {
+		return err
+	}
+	c.Name = info.Name
+	return nil
+}
+
+// ErasureInfo hols erasure coding and bitrot related information.
+type ErasureInfo struct {
+	Algorithm         string
+	DataBlocks        int
+	ParityBlocks      int
+	BlockSize         int64
+	Index             int
+	Distribution      []int
+	ChecksumAlgorithm BitrotAlgorithm
+	Checksums         []ChecksumInfo
+}
+
+// MarshalJSON marshals the ChecksumInfo struct
+func (e ErasureInfo) MarshalJSON() ([]byte, error) {
+	type erasureinfo struct {
+		Algorithm         string         `json:"algorithm"`
+		DataBlocks        int            `json:"data"`
+		ParityBlocks      int            `json:"parity"`
+		BlockSize         int64          `json:"blockSize"`
+		Index             int            `json:"index"`
+		Distribution      []int          `json:"distribution"`
+		ChecksumAlgorithm string         `json:"bitrot"`
+		Checksums         []ChecksumInfo `json:"checksum,omitempty"`
+	}
+
+	if !e.ChecksumAlgorithm.Available() {
+		return nil, errBitrotHashAlgoInvalid
+	}
+
+	info := erasureinfo{
+		Algorithm:         e.Algorithm,
+		DataBlocks:        e.DataBlocks,
+		ParityBlocks:      e.ParityBlocks,
+		BlockSize:         e.BlockSize,
+		Index:             e.Index,
+		Distribution:      e.Distribution,
+		ChecksumAlgorithm: e.ChecksumAlgorithm.String(),
+		Checksums:         e.Checksums,
+	}
+	return json.Marshal(info)
+}
+
+// UnmarshalJSON unmarshals the the given data into the ChecksumInfo struct
+func (e *ErasureInfo) UnmarshalJSON(data []byte) error {
+	type erasureinfo struct {
+		Algorithm         string         `json:"algorithm"`
+		DataBlocks        int            `json:"data"`
+		ParityBlocks      int            `json:"parity"`
+		BlockSize         int64          `json:"blockSize"`
+		Index             int            `json:"index"`
+		Distribution      []int          `json:"distribution"`
+		ChecksumAlgorithm string         `json:"bitrot"`
+		Checksums         []ChecksumInfo `json:"checksum,omitempty"`
+	}
+
+	var info erasureinfo
+	if err := json.Unmarshal(data, &info); err != nil {
```

Comment:
Won't... Unmarshaling is done in [here](https://github.com/minio/minio/pull/4758/files#diff-52ef863c8157cf06131c2b4b684d6d3eR153-R196) - This is just to provide symmetry (and for testing)